crates/rust-analyzer/src/config.rs

```diff
@@ -282,6 +282,8 @@ config_data! {
                 "scope": "expr"
             }
         }"#,
+        /// Whether to enable term search based snippets like `Some(foo.bar().baz())`.
+        completion_term_search_enable: bool = "true",
```

Comment:
Let's put this to false by default and add a warning that this may slow down completions. Completions are already rather slow atm so I don't feel like it's a good idea to enable this by default right now. It also potentially clutters the completions with things.

Reply:
Also, the key should probably be `completion_termSearch_enable`